### ISSUE TITLE
Add support for jellyfin 10.9

### ISSUE THREAD

**Issue by foux** [firecore/InfuseSync#24]
Hi there.
Jellyfin 10.9 just made its way out of beta. Plugin compatibility would be nice!

Here are the erros loading the plugin on 10.9 : 
```
2024-05-11 21:46:10.556 +02:00] [ERR] [1] Emby.Server.Implementations.Plugins.PluginManager: Failed to load assembly "/config/data/plugins/InfuseSync_1.4.2.0/InfuseSync.dll". This error occurs when a plugin references an incompatible version of one of the shared libraries. Disabling plugin
System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types.
Could not load type 'MediaBrowser.Controller.Plugins.IServerEntryPoint' from assembly 'MediaBrowser.Controller, Version=10.9.0.0, Culture=neutral, PublicKeyToken=null'.
Could not load type 'MediaBrowser.Controller.Plugins.IServerEntryPoint' from assembly 'MediaBrowser.Controller, Version=10.9.0.0, Culture=neutral, PublicKeyToken=null'.
Could not load file or assembly 'SQLitePCL.pretty, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeModule.GetTypes()
   at Emby.Server.Implementations.Plugins.PluginManager.LoadAssemblies()+MoveNext()
System.TypeLoadException: Could not load type 'MediaBrowser.Controller.Plugins.IServerEntryPoint' from assembly 'MediaBrowser.Controller, Version=10.9.0.0, Culture=neutral, PublicKeyToken=null'.
System.TypeLoadException: Could not load type 'MediaBrowser.Controller.Plugins.IServerEntryPoint' from assembly 'MediaBrowser.Controller, Version=10.9.0.0, Culture=neutral, PublicKeyToken=null'.
System.IO.FileNotFoundException: Could not load file or assembly 'SQLitePCL.pretty, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

File name: 'SQLitePCL.pretty, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'
```

Thanks

**Comment by brettwilcox** [comment]
I also wanted to point out this fix by [jet082](https://github.com/jet082). I've not tested, but may be a good place to start.

- https://github.com/firecore/InfuseSync/issues/21#issuecomment-1763167730

> So, taking the SQLitePCL.pretty dll file from the latest version (10.8.11) here (https://repo.jellyfin.org/releases/server/portable/stable/) and placing it in the same plugin directory as InfuseSync allows it to work again.

> Perhaps all that needs happening is bundling this dll when installing via infuse.

**Comment by Oatavandi** [comment]
Does not work with jellyfin 10.9.1

**Comment by simoncaron** [comment]
For anyone interested, until this is resolved, [I did the minimum fixes](https://github.com/simoncaron/InfuseSync/commit/51d5bba7fab3e701c313fa3dce16095caf7dff09) to get it working again. Didn't rewrite the SQL part so it's still relying on SQLitePCL.

I've compiled the DLL and uploaded it (along with the SQL one) here: https://github.com/simoncaron/InfuseSync/releases/tag/fix-jellyfin-10.9.x

**Comment by russelg** [comment]
@jet082 you need to edit the meta.yml file for the plugin, update the abiVersion to "10.9.0.0". Should work after that.

**Comment by russelg** [comment]
What did you do with the new DLL files? They are meant to go in your jellyfin's plugin directory. I don't know what your paths are like, but it'll be <wherever your jellyfin config is>/plugins/InfuseSync_1.4.2.0. Put the new DLLs in there, replacing any old ones, then edit the meta file there as well. Restart and should work.

**Comment by russelg** [comment]
> I've verified via the md5sums that what is loaded is indeed the exact one on @simoncaron's releases page. Moreover, I've looked at the `meta.json` file and it has ` "targetAbi": "10.9.0.0",` so I think that it's something else.

Not sure what to tell you, if you look at the diff for simoncaron's fixes, you can see `IServerEntryPoint` was replaced with `IHostedService`. There are no references to `IServerEntryPoint` in the fork anymore. I also downloaded the DLL and decompiled it, can't see any references to it there either. If it's loading the correct DLLs, there should not be errors crying about `IServerEntryPoint`.

**Comment by russelg** [comment]
Is your jellyfin version definitely 10.9.1? Those assemblies are part of Jellyfin so the only way they wouldn't be present is if you're not actually on 10.9.1.

I have `[INF] [1] Main: Jellyfin version: 10.9.1` in my log for example.

**Comment by russelg** [comment]
Yep, looked into it a bit more, seems unstable is using 10.9.0.0 as the version for the assemblies still. I just downloaded a fresh copy of stable 10.9.1, and the assemblies there are versioned 10.9.1.0. Quite odd...

**Comment by EthanBezz** [comment]
The devs are aware of the issue. Fixed version will be "available soon".

https://community.firecore.com/t/add-infusesync-plugin-support-for-jellyfin-10-9/49006/7

**Comment by Cptnobeard** [comment]
is there any update on this issue and maybe a aprox release date?
anyway i want to thank everyone for there hard work!

**Comment by firecore** [comment]
We have just posted a [beta version of 1.5.0](https://github.com/firecore/InfuseSync/releases/tag/v1.5.0) which is available to test.

We are still doing some final internal testing but things are looking good.

**Comment by hostler** [comment]
> Any release date available yet?

It's released since a few days.

When will it be in the Jellyfin repo?

**Comment by firecore** [comment]
We've released the [1.5.0 version of InfuseSync](https://github.com/firecore/InfuseSync/releases/tag/v1.5.0) which adds support for Jellyfin 10.9.

We're working with the Jellyfin admins to get this new version added to the main Jellyfin plugin catalog, but in the meantime you can install the update by following the [steps in this post on our forum](https://community.firecore.com/t/add-infusesync-plugin-support-for-jellyfin-10-9/49006/22).
